internal/task/queue.go

```diff
@@ -130,7 +130,7 @@ func (t *Queue) worker() {
 			klog.ErrorS(err, "requeuing", "key", item.Key)
 			t.queue.AddRateLimited(Element{
 				Key:       item.Key,
-				Timestamp: time.Now().UnixNano(),
+				Timestamp: 0,
 			})
 		} else {
```

Comment:
Yes, I think it's also a way to solve the removal of the previously enqueued element. But we have another problem, that the rate limiter will not work as we expected (an exponential backoff) because the enqueued element is changing.

I've also read some implementation in kubernetes, for example, https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpoint/endpoints_controller.go#L373. It will enqueue only the key string, along with some max retries logic.

I think the best solution is to reimplement the RateLimitingQueue (https://github.com/kubernetes/client-go/blob/master/util/workqueue/rate_limiting_queue.go) to use item.Key for the ratelimiter, and enqueue the whole element to the work queue. But it could add more complexity to the code, so IMHO this is the simplest way to achieve the best effect.

Reply:
Oh, now I understand. Thank you for clarifying.  Then maybe an option could be to accept this change as it is, to fix the problem right away. But leaving an issue open to track this need of refactoring, so that someone can make this improvement in the future.

Reply:
agreed. We can merge this, and verify after this some problematic increase of memory usage, or even have a follow up to reimplement the logic, but so far I don't want to block improvements on performance :)

Thanks @lyt99